### PR TITLE
Enhencements about reliability and new feature for create clause

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
This patch add some reliability enhencements including:
* put namespace and table name in HBase together when create table, ignore duplicated tableName. 
* add verification with properties in Option and data type supported.
* add column name verification when alterdrop/alteradd column.
* handle null value, rows that has more/less column than expected.
  * If number of value is less than expected, fill with null;
  * If number of value is more than expected, cut the rest out;
  * If type of input value doesn't match what is expected, replace with null
  * If there is null value in RowKey, skip this row;

* add keyword splitKeys in create clause to init regions while creating tables in HBase

### CODE REVIEW

**Comment by WangTaoTheTonic** [comment]
In https://github.com/WangTaoTheTonic/astro/commit/016a523fe91478d0f7014f309dcf458117ca0225 we add keyword `splitKeys` in create clause to init regions while creating tables in HBase.

**Comment by WangTaoTheTonic** [comment]
test this please.

File: src/main/scala/org/apache/spark/sql/hbase/HBaseRelation.scala [hunk 1]
```diff
@@ -117,8 +135,31 @@ class HBaseSource extends SchemaRelationProvider {
...
+        // We can not use ";" as delimiter, as SQL clause use it as terminating symbol.
+        splitKeysInfo.split(groupDelimiter).map(_.trim).map { splitKeyStr: String =>
+          val splitKeyArr = splitKeyStr.split(dataDelimiter).map(_.trim)
+          if (splitKeyArr.length != dataTypes.length) {
+            throw new Exception(s"There are ${dataTypes.length} keys, " +
+              s"but have ${splitKeyArr.length} split key values")
+          }
+          val rowArr: Array[Any] = splitKeyArr.zipWithIndex.map {
+            case (skv, idx) => Util.getValueFromString(skv, dataTypes(idx))
+          }
```
Comment (ravipesala) [inline src/main/scala/org/apache/spark/sql/hbase/HBaseRelation.scala]: Can you add some test cases for splitKeys

**Comment by WangTaoTheTonic** [comment]
added a test case for splitKeys in `TestBaseWithSplitData`, which will check if region number of created table is same as we specified in create.
test this pleas.

**Comment by WangTaoTheTonic** [comment]
test this please

**Comment by ravipesala** [comment]
LGTM Thanks for working on this. Merged to master

**Pull request merged**